plugins/rest/src/authenticator.cpp

```diff
@@ -0,0 +1,185 @@
+//    _   _____   __________
+//   | | / / _ | / __/_  __/     Visibility
+//   | |/ / __ |_\ \  / /          Across
+//   |___/_/ |_/___/ /_/       Space and Time
+//
+// SPDX-FileCopyrightText: (c) 2022 The VAST Contributors
+// SPDX-License-Identifier: BSD-3-Clause
+
+#include "rest/authenticator.hpp"
+
+#include <vast/detail/base64.hpp>
+
+#include <caf/scoped_actor.hpp>
+
+#include <array>
+
+// TODO: Investigate if this is available on mac
+#include <sys/random.h>
```

Comment:
Is there a reason not to use the C++ standard library tools here?

Reply:
Fair point. I think we already link against OpenSSL via CAF, so maybe we can use that for it?

Reply:
Good point, recent version of OpenSSL use `getentropy()` on Linux anyways to implement `RANDOM_bytes()`